Assignment1/src/main/java/org/christmas/lights/grid/ChristmasLightGridImpl.java

```diff
@@ -0,0 +1,105 @@
+package org.christmas.lights.grid;
+/**
+ * ChristmasLightGridImpl 
+ * @author narec
+ *
+ */
+public class ChristmasLightGridImpl implements ChristmasLightGrid{
+	 private Light[][] grid;
+	 private LightOnCounter lightCounter;
+	/**
+	  * create a light grid for given coordinates
+	  * @param rows
+	  * @param columns
+	  */
+	 public ChristmasLightGridImpl(int rows, int columns) {
+	     grid = new Light[rows][columns];
+	     initializeGrid();
+	     lightCounter=new LightOnCounterImpl();
+	 }
+/**
+ * initialize the grid 
+ */
+	 private void initializeGrid() {
+	     for (int i = 0; i < grid.length; i++) {
+	         for (int j = 0; j < grid[i].length; j++) {
+	             grid[i][j] = new ChristmasLight();
+	         }
+	     }
+	 }
+	 
+	 /**
+	  * toggle light for given coordinates
+	  * @param row
+	  * @param column
+	  */
+
+	 public void toggleLight(int row, int column) {
+	     if (isValidCoordinate(row, column)) {
+	    	 Light light = grid[row][column];
+	         if (light != null) {
+	             if (light.isOn()) {
+	                 light.turnOff();
+	                 lightCounter.decrementLightOnCount();
```

Comment:
The logic for the following is wrong . 

it should be eg.

 `for (int i = startX; i < endX; i++) {
            for (int j = startY; j < endY; j++) {
                grid[i][j].turnOff();
            }
        }`

Reply:
Fixed